src/testing/flow_solver_cases/naca0012.cpp

```diff
@@ -0,0 +1,95 @@
+//=========================================================
+// NACA0012
+//=========================================================
+template <int dim, int nstate>
+NACA0012<dim, nstate>::NACA0012(const PHiLiP::Parameters::AllParameters *const parameters_input)
+        : FlowSolverCaseBase<dim, nstate>(parameters_input)
+{
+}
+
+template <int dim, int nstate>
+void NACA0012<dim,nstate>::display_flow_solver_setup() const
+{
+    using PDE_enum = Parameters::AllParameters::PartialDifferentialEquation;
+    const PDE_enum pde_type = this->all_param.pde_type;
+    const double pi = atan(1.0) * 4.0;
+    std::string pde_string;
+    if (pde_type == PDE_enum::euler)                {pde_string = "euler";}
+    if (pde_type == PDE_enum::navier_stokes)        {pde_string = "navier_stokes";}
+    this->pcout << "- PDE Type: " << pde_string << std::endl;
+    this->pcout << "- Polynomial degree: " << this->all_param.grid_refinement_study_param.poly_degree << std::endl;
+    this->pcout << "- Courant-Friedrich-Lewy number: " << this->all_param.flow_solver_param.courant_friedrich_lewy_number << std::endl;
+    this->pcout << "- Freestream Reynolds number: " << this->all_param.navier_stokes_param.reynolds_number_inf << std::endl;
+    this->pcout << "- Freestream Mach number: " << this->all_param.euler_param.mach_inf << std::endl;
+    this->pcout << "- Angle of attack: " << this->all_param.euler_param.angle_of_attack*180/pi << std::endl;
+    this->pcout << "- Side-slip angle: " << this->all_param.euler_param.side_slip_angle*180/pi << std::endl;
+}
+
+template <int dim, int nstate>
+std::shared_ptr<Triangulation> NACA0012<dim,nstate>::generate_grid() const
+{
+    //Dummy triangulation
+    std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation>(MPI_COMM_WORLD);
+    dealii::GridGenerator::Airfoil::AdditionalData airfoil_data;
+    dealii::GridGenerator::Airfoil::create_triangulation(*grid, airfoil_data);
+    grid->refine_global();
+    return grid;
+}
+
+template <int dim, int nstate>
+void NACA0012<dim,nstate>::set_higher_order_grid(std::shared_ptr<DGBase<dim, double>> dg) const
+{
+    const std::string mesh_filename = this->all_param.flow_solver_param.input_mesh_filename+std::string(".msh");
+    std::shared_ptr<HighOrderGrid<dim,double>> naca0012_mesh = read_gmsh<dim, dim> (mesh_filename);
+    dg->set_high_order_grid(naca0012_mesh);
+    for (unsigned int i=0; i<this->all_param.grid_refinement_study_param.num_refinements; ++i) {
+        dg->high_order_grid->refine_global();
+    }
+    const unsigned int n_global_active_cells = dg->triangulation->n_global_active_cells();
+    const unsigned int n_dofs = dg->dof_handler.n_dofs();
+    this->pcout << "Dimension: " << dim << "\t Polynomial degree p: " << this->all_param.grid_refinement_study_param.poly_degree << std::endl
+          << ". Number of active cells: " << n_global_active_cells
+          << ". Number of degrees of freedom: " << n_dofs
+          << std::endl;
```

Comment:
It's a bit inconsistent that you're defining variables in some places (`n_dofs` and `n_global_active_cells`) but not in other places (`all_param.grid_refinement_study_param.num_refinements`, for instance).

Reply:
Fixed in recent commit.